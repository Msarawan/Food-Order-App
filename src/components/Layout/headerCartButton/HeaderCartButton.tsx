import * as React from 'react';
import { useContext } from 'react';
import CartIcon from '../../Cart/carticon/CartIcon';
import CartContext from '../../../store/CartContext';
import classes from './HeaderCartButton.module.css';


interface IHeaderCartButtonProps {
  onClick:any
}
const HeaderCartButton: React.FC<IHeaderCartButtonProps> = ({onClick}) => {
  const cartCtx = useContext(CartContext);

  const numberOfCartItems = cartCtx.items.reduce((curNumber, item) => {
    return curNumber + item['amount'];
  }, 0);

  return (
    <button className={classes.button}  onClick={onClick}>
      <span className={classes.icon}>
        <CartIcon />
      </span>
      <span className={classes.name}>Your Cart</span>
      <span className={classes.badge}>{numberOfCartItems}</span>
    </button>
  );
};

export default HeaderCartButton;