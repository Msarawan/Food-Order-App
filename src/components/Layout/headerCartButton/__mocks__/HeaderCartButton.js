import React from 'react';
const MockedHeaderCartButton = () => {
  return <></>;
}
export default MockedHeaderCartButton;