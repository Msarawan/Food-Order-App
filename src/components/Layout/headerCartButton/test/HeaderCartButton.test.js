import 'jsdom-global/register';
import React from 'react';
import HeaderCartButton from "../../headerCartButton/HeaderCartButton";
//import Adapter from 'enzyme-adapter-react-16';
import {mount,shallow} from "enzyme";

describe("Food Order app testing", () => {
  let wrapper;
  beforeEach(() => {
   wrapper = mount(<HeaderCartButton/>);
  });

  test("render the cart name", () => {
    expect(wrapper.find(".name").text()).toContain("Your Cart");
  });

  expect(wrapper).toMatchSnapshot();
 
})

describe('render a button on onclick event' , () =>{
  const showCartHandler = jest.fn();
  const  wrapper = shallow(<HeaderCartButton onClick = {showCartHandler}/>)

  const element = wrapper.find('button').first();
  //const fn = jest.fn()
  element.simulate('click')
  expect(showCartHandler).toHaveBeenCalledTimes(1)
  //console.log(showCartHandler);
 expect(wrapper).toMatchSnapshot();
})