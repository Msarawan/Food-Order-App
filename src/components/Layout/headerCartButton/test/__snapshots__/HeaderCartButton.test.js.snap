// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `undefined`;

exports[` 2`] = `
<button
  className="button"
  onClick={
    [MockFunction] {
      "calls": [
        [],
      ],
      "results": [
        {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <span
    className="icon"
  >
    <CartIcon />
  </span>
  <span
    className="name"
  >
    Your Cart
  </span>
  <span
    className="badge"
  >
    0
  </span>
</button>
`;
